src/i2c.rs

```diff
@@ -830,6 +841,7 @@ where
 
             // If we received a NACK, then this is an error
             if sr1.af().bit_is_set() {
+                self.i2c.sr1.modify(|_, w| w.af().clear_bit());
```

Comment:
Same question as above. If guess we should also factor this out into its own function to make it clearer what it does.

Reply:
Should now be refactored into `check_and_clear_errors()`